.github/workflows/build_and_deploy.yml

```
@@ -0,0 +1,75 @@
+# Copyright (c) 2022 ETH Zurich
+#
+# SPDX-License-Identifier: BSL-1.0
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+name: Build, test and deploy docker images
+
+# The builds are triggered with pushes on the 'main' branch or with a pull
+# request against the 'main' branch
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+env:
+  base_version: 1
+  base_image_name: pika-ci-base
+  hip_version: 1
+  hip_image_name: pika-ci-hip
+
+jobs:
+  # Build base docker image
+  build_and_deploy:
+    # Cannot use the env variables in the name
+    name: Build and deploy images
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USER }}
+          password: ${{ secrets.DOCKER_PASS }}
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      # Base image #
+      - name: Build base image
+        if: ${{ github.event != 'push' }}
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: Dockerfile.base
+          push: false
```

Sorry for one more comment, but I just realized: can this be 
```suggestion
          push: ${{ github.event == 'push' }}
```
? If that is legal, then you'd also get rid of the duplication between deploy and non-deploy.

---

Definitely, I will try that out thanks!